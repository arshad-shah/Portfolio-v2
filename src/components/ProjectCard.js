import React from "react";
import { styled } from "@mui/material/styles";
import Card from "@mui/material/Card";
import CardActions from "@mui/material/CardActions";
import CardContent from "@mui/material/CardContent";
import CardHeader from "@mui/material/CardHeader";
import { Chip, IconButton, Tooltip, Typography } from "@mui/material";
import { FiGithub, FiExternalLink } from "react-icons/fi";
import {
  SiCss3,
  SiHtml5,
  SiJava,
  SiJavascript,
  SiJquery,
  SiKotlin,
  SiMaterialui,
  SiNextdotjs,
  SiReact,
  SiRedux,
  SiSqlite,
  SiStyledcomponents,
  SiTailwindcss,
  SiGoogleplay,
} from "react-icons/si";
import PropTypes from "prop-types";
const PREFIX = "ProjectCard";

const classes = {
  root: `${PREFIX}-root`,
  githubLinksAnchor: `${PREFIX}-githubLinksAnchor`,
  LanguageChip: `${PREFIX}-LanguageChip`,
};

const StyledCard = styled(Card)(({ theme }) => ({
  [`&.${classes.root}`]: {
    backgroundColor: "#ffffff",
    padding: "1rem",
  },

  [`& .${classes.githubLinksAnchor}`]: {
    "&:visited": {
      color: "#000000",
    },
    "&:link": {
      color: "#000000",
    },
  },

  [`& .${classes.LanguageChip}`]: {
    margin: "0.3rem",
    borderRadius: "5px",
    padding: "0.5rem",
    fontWeight: "bold",
  },
}));

export default function ProjectCard({
  heading,
  description,
  languages,
  linkAriaLabel,
  repoLink,
  siteLink,
  shopLink,
}) {
  //a function that find the correct icon for a given language array
  const getIcon = (language) => {
    switch (language) {
      case "Kotlin":
        return <SiKotlin />;
      case "React Native":
      case "ReactJS":
        return <SiReact />;
      case "Redux toolkit":
        return <SiRedux />;
      case "Tailwind CSS":
        return <SiTailwindcss />;
      case "NextJS":
        return <SiNextdotjs />;
      case "Material UI":
        return <SiMaterialui />;
      case "Styled components":
        return <SiStyledcomponents />;
      case "HTML":
        return <SiHtml5 />;
      case "CSS":
        return <SiCss3 />;
      case "Javascript":
        return <SiJavascript />;
      case "JQuery":
        return <SiJquery />;
      case "Java":
        return <SiJava />;
      case "SQLite":
        return <SiSqlite />;
      default:
        return null;
    }
  };

  return (
    <StyledCard raised className={classes.root}>
      <CardHeader id="heading" title={heading} data-testid="projectHeading" />
      <CardContent>
        <Typography variant="body2">{description}</Typography>
        <Typography variant="overline">
          <br />
          {languages.map((language) => (
            <Chip
              icon={getIcon(language)}
              color="secondary"
              key={language}
              label={language}
              className={classes.LanguageChip}
              data-testid="languageChip"
            />
          ))}
        </Typography>
      </CardContent>
      <CardActions>
        {/* Link to github */}
        <Tooltip title="Github Repo" arrow>
          <IconButton
            data-testid="githublink"
            data-block="projectlink"
            role="button"
            color="secondary"
            aria-label={linkAriaLabel}
            href={repoLink}
            size="large"
          >
            <FiGithub />
          </IconButton>
        </Tooltip>

        {shopLink && (
          <Tooltip title="Play Store Link" arrow>
            <IconButton
              data-testid="shoplink"
              data-block="shoplink"
              role="button"
              color="secondary"
              aria-label={linkAriaLabel}
              href={shopLink}
              size="large"
            >
              <SiGoogleplay />
            </IconButton>
          </Tooltip>
        )}

        {siteLink ? (
          <Tooltip title="Live Site Link" arrow>
            <IconButton
              data-testid="externalSiteLink"
              data-block="projectLinkToSite"
              role="button"
              color="secondary"
              aria-labelledby={heading}
              href={siteLink}
              size="large"
            >
              <FiExternalLink />
            </IconButton>
          </Tooltip>
        ) : null}
      </CardActions>
    </StyledCard>
  );
}

ProjectCard.propTypes = {
  heading: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  languages: PropTypes.arrayOf(PropTypes.string).isRequired,
  linkAriaLabel: PropTypes.string.isRequired,
  repoLink: PropTypes.string.isRequired,
  siteLink: PropTypes.string,
  shopLink: PropTypes.string,
};
