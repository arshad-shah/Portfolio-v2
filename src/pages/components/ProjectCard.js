import React from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardHeader from "@material-ui/core/CardHeader";
import useStyles from "./ProjectCard.styles";
import { IconButton, Typography } from "@material-ui/core";
import GitHubIcon from "@material-ui/icons/GitHub";
import ShopRoundedIcon from "@material-ui/icons/ShopRounded";

export default function ProjectCard({
  heading,
  description,
  languages,
  linkAriaLabel,
  repoLink,
}) {
  const classes = useStyles();
  return (
    <Card raised className={classes.root}>
      <CardHeader title={heading} />
      <CardContent>
        <Typography variant="body2">{description}</Typography>
        <Typography variant="overline">
          Languages Used: <br />
          {languages}
        </Typography>
      </CardContent>
      <CardActions>
        {/* Link to github */}
        <IconButton role="button" color="default" aria-label={linkAriaLabel}>
          <a className={classes.githubLinksAnchor} href={repoLink}>
            {heading === "Android App Project | Nimaz" ? (
              <ShopRoundedIcon />
            ) : (
              <GitHubIcon />
            )}
          </a>
        </IconButton>
      </CardActions>
    </Card>
  );
}
