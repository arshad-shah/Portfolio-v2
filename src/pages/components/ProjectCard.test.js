import React from "react";
import { render, screen } from "@testing-library/react";
import ProjectCard from "./ProjectCard";

describe("ProjectCard of the site:", () => {
  const data = [
    {
      heading: "Portfolio website version 1.0",
      description:
        "A Portfolio site that contains all the work i have done so far. Includes work from college such as exercises, projects and also from outside college such as the android app.This is the first version written in pure HTML, CSS and javascript with a pinch of jquery.",
      languages: "HTML | CSS | Javascript | jQuery",
      linkAriaLabel: "link to Github for portfolio version 1",
      repoLink: "https://github.com/arshad-shah/arshad-shah.github.io",
    },
  ];

  it("should render ProjectCard with all the information in it", () => {
    const { container } = render(<ProjectCard {...data} />);
    expect(container).toMatchSnapshot();
  });
});
