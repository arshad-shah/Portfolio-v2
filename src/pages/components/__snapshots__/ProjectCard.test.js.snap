// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard of the site: should render ProjectCard with all the information in it 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root ProjectCard-root-1 MuiPaper-elevation8 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
      data-testid="projectHeading"
      id="heading"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Portfolio website version 1.0
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        A Portfolio site that contains all the work i have done so far. Includes work from college such as exercises, projects and also from outside college such as the android app.This is the first version written in pure HTML, CSS and javascript with a pinch of jquery.
      </p>
      <span
        class="MuiTypography-root MuiTypography-overline"
      >
        Languages Used: 
        <br />
        HTML | CSS | Javascript | jQuery
      </span>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        aria-disabled="false"
        aria-label="link to Github for portfolio version 1"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
        data-block="projectlink"
        href="https://github.com/arshad-shah/arshad-shah.github.io"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        aria-labelledby="Portfolio website version 1.0"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
        data-block="projectLinkToSite"
        href="https://arshad-shah.github.io/portfoliov1/index.html"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 19H6c-.55 0-1-.45-1-1V6c0-.55.45-1 1-1h5c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.11 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-6c0-.55-.45-1-1-1s-1 .45-1 1v5c0 .55-.45 1-1 1zM14 4c0 .55.45 1 1 1h2.59l-9.13 9.13c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L19 6.41V9c0 .55.45 1 1 1s1-.45 1-1V3h-6c-.55 0-1 .45-1 1z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
